ui/app/AppLayouts/Wallet/controls/TokenSelectorNew.qml

```diff
@@ -0,0 +1,162 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import StatusQ.Components 0.1
+import StatusQ.Components.private 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1
+
+import AppLayouts.Wallet.panels 1.0
+
+import utils 1.0
+
+Control {
+    id: root
+
+    /** Expected model structure: see TokenSelectorPanel::assetsModel **/
+    property alias assetsModel: tokenSelectorPanel.assetsModel
+
+    /** Expected model structure: see TokenSelectorPanel::collectiblesModel **/
+    property alias collectiblesModel: tokenSelectorPanel.collectiblesModel
+
+    signal assetSelected(string key)
+    signal collectionSelected(string key)
+    signal collectibleSelected(string key)
+
```

Comment:
I'm going to add that part in a next step if needed. There are also some other cases to be taken into account - e.g. setting token that's not present in a model (when opening from activities history). So I will add what's needed a bit later when knowing more about those cases.

Reply:
It will be needed in case of the SwapModal, we know that already :)

Reply:
Yeah, I know but let me do it in a next step while integrating to send/swap because I'm not sure if it's better to parametrise or decompose even more into independent parts and build those two variants as independent compositions.